pkg/stream/producer_daemon.go

```diff
@@ -170,6 +187,23 @@ func NewProducerDaemonWithInterfaces(logger log.Logger, metric metric.Writer, ag
 	}
 }
 
+func (d *producerDaemon) IsStopped() bool {
+	return atomic.LoadInt32(&d.stopped) != 0
+}
+
+func (d *producerDaemon) Restart() {
+	d.lck.Lock()
+	defer d.lck.Unlock()
+
+	if d.outCh.IsClosed() {
+		d.outCh = NewOutputChannel(d.logger, d.settings.BufferSize)
+	}
+
+	if atomic.LoadInt32(&d.stopped) != 0 {
```

Comment:
You don't need to check for != 0 here - writing a 0 to a variable holding 0 is fine.

While it isn't a problem here, this pattern additionally just feels like a race condition as you have a race between the check and the actual use later on